src/test/java/com/example/moviemanager/service/TmdbClientTest.java

```diff
@@ -0,0 +1,155 @@
+                .willReturn(aResponse()
+                        .withHeader(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON_VALUE)
+                        .withStatus(HttpStatus.OK.value())
+                        .withBody(objectMapper.writeValueAsString(
+                                new MovieDetailsResponse(List.of(new MovieDetails("GOT Tv-show.", 8.4)))
+                        ))
+                )
+        );
+
+        // when
+        String actualMovieDetailsResponse = objectMapper.writeValueAsString(tmdbClient.find(findMovieInfo));
+        System.out.println(actualMovieDetailsResponse);
+
+        // then
+        JSONAssert.assertEquals(
+                """
+                          {   "results": [
+                                   {
+                                       "overview": "GOT Tv-show.",
+                                       "vote_average": 8.4
+                                   }
+                              ]
+                          }
+                        """, actualMovieDetailsResponse, true);
+    }
+
+    @Test
+    public void returns_movie_response_according_to_given_title_and_release_year() throws JsonProcessingException, JSONException {
+        // given
+        FindMovieInfo findMovieInfo = new FindMovieInfo("Home Alone", MovieType.MOVIE, 1990);
+        wireMockServer.stubFor(get(MOVIES_URL + buildQuery(findMovieInfo))
+                .willReturn(aResponse()
+                        .withHeader(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON_VALUE)
+                        .withStatus(HttpStatus.OK.value())
+                        .withBody(objectMapper.writeValueAsString(
+                                new MovieDetailsResponse(List.of(new MovieDetails("Christmas movie.", 7.4)))
+                        ))
+                )
+        );
+
+        // when
+        String actualMovieDetailsResponse = objectMapper.writeValueAsString(tmdbClient.find(findMovieInfo));
+
+        // then
+        JSONAssert.assertEquals(
+                """
+                          {   "results": [
+                                   {
+                                       "overview": "Christmas movie.",
+                                       "vote_average": 7.4
+                                   }
+                              ]
+                          }
+                        """, actualMovieDetailsResponse, true);
+    }
+
+    @ParameterizedTest
+    @CsvFileSource(resources = "/invalid_find_movie_parameters.csv", numLinesToSkip = 1)
+    public void returns_empty_response_when_provided_non_existing_movie_parameters(
+            String title, MovieType type, int releaseYear
+    ) throws JsonProcessingException, JSONException {
+        // given
+        FindMovieInfo findMovieInfo = new FindMovieInfo(title, type, releaseYear);
+        String url = type.equals(MovieType.MOVIE) ? MOVIES_URL : TV_SHOWS_URL;
```

Comment:
Instead of this, I suggest replacing MovieType in csv with actual URL. Since it is given part, you should not have any logic in it. It should be plain facts.